test/integration/sagemaker/test_smdataparallel.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import os
+
+import pytest
+import sagemaker
+from sagemaker.tensorflow import TensorFlow
+from sagemaker.utils import unique_name_from_base
+
+from integration import DEFAULT_TIMEOUT, RESOURCE_PATH
+from integration.sagemaker.timeout import timeout
+
+
+@pytest.mark.skip_cpu
+@pytest.mark.skip_generic
+@pytest.mark.parametrize(
+    "instances, instance_type",
+    [(2, "ml.p3.16xlarge")],
+)
+def test_smdataparallel_training(instances, instance_type, sagemaker_session, image_uri, framework_version, tmpdir):
+    default_bucket = sagemaker_session.default_bucket()
+    output_path = "s3://" + os.path.join(default_bucket, "tensorflow/smdataparallel")
```

Comment:
although it doesn't matter for the tests in this platform, we should avoid use of `os.path.join` for s3 path construction since s3 paths always use `/` while `os.path` uses the platform's delimiter, which may be `\`

Reply:
Very good point!